### PR TITLE
Add fixes for ThemeData

### PR DESCRIPTION
**Pull request by Piinks**
This adds flutter/dart fixes for https://github.com/flutter/flutter/pull/81336 which deprecated several properties of ThemeData
There is a full migration guide available at https://flutter.dev/docs/release/breaking-changes/theme-data-accent-properties

Part of https://github.com/flutter/flutter/issues/74908

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test_fixes/material.dart.expect [hunk 1]
```diff
@@ -303,6 +303,41 @@ void main() {
   renderObjectToWidgetElement.moveRenderObjectChild(renderObject, object);
   renderObjectToWidgetElement.removeRenderObjectChild(renderObject);
 
+  // Changes made in https://github.com/flutter/flutter/pull/81336
+  ThemeData themeData = ThemeData();
+  themeData = ThemeData(colorScheme: ColorScheme(secondary: Colors.red));
```
[the remark this answers is not in the document]
Comment (HansMuller) [reply packages/flutter/test_fixes/material.dart.expect:308]: Another scenario:

```dart
ThemeData(
  primaryColor: Colors.green,
  accentColor: Colors.red,
)
// expect
ThemeData(
  primaryColor: Colors.green,
).copyWith(
  secondaryColor: Colors.red
)
```

[the remark this answers is not in the document]
Comment (Piinks) [reply packages/flutter/test_fixes/material.dart.expect:308]: > Another scenario:
> 
> ```dart
> ThemeData(
>   primaryColor: Colors.green,
>   accentColor: Colors.red,
> )
> // expect
> ThemeData(
>   primaryColor: Colors.green,
> ).colorScheme.copyWith(
>   secondaryColor: Colors.red,
> )
> ```

[the remark this answers is not in the document]
Comment (HansMuller) [reply packages/flutter/test_fixes/material.dart.expect:308]: Oops, yes EXACTLY

[the remark this answers is not in the document]
Comment (Piinks) [reply packages/flutter/test_fixes/material.dart.expect:308]: These are much more helpful. I will update. :)

**Comment by Piinks** [comment]
@HansMuller PTAL. :)

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**